### PR TITLE
Upgrading to latest version of eslint

### PR DESCRIPTION
**Pull request by johnsimons**
The latest version of `eslint` has changed the way it is configured. It now uses [flat config](https://eslint.org/blog/2023/10/flat-config-rollout-plans/)

### CODE REVIEW

File: src/Frontend/.eslintrc.cjs [hunk 1]
```diff
@@ -1,33 +0,0 @@
...
-  extends: ["plugin:vue/vue3-essential", "eslint:recommended", "plugin:@typescript-eslint/recommended", "@vue/eslint-config-prettier/skip-formatting"],
-  parser: "vue-eslint-parser",
-  plugins: ["@typescript-eslint", "github", "import", "unused-imports", "prefer-arrow"],
-  parserOptions: {
-    ecmaVersion: "latest",
-    parser: "@typescript-eslint/parser",
-    extraFileExtensions: [".vue"],
-  },
-  rules: {
-    "prettier/prettier": "error",
```
Comment (johnsimons) [inline src/Frontend/.eslintrc.cjs:20]: replaced by eslint-config-prettier

File: src/Frontend/.eslintrc.cjs [hunk 2]
```diff
@@ -1,33 +0,0 @@
...
-    "require-await": "error",
-    "no-await-in-loop": "error",
-    "github/no-then": "error",
```
Comment (johnsimons) [inline src/Frontend/.eslintrc.cjs:23]: couldn't find repacement

File: src/Frontend/.eslintrc.cjs [hunk 3]
```diff
@@ -1,33 +0,0 @@
...
-    "prefer-rest-params": "error",
-    "prefer-spread": "error",
-    "no-var": "error",
-    "prefer-const": "error",
-    eqeqeq: ["error", "smart"],
-    "unused-imports/no-unused-imports": "error",
-    "import/no-duplicates": "error",
```
Comment (johnsimons) [inline src/Frontend/.eslintrc.cjs:30]: couldn't find replacement

File: src/Frontend/package.json [hunk 4]
```diff
@@ -9,62 +9,59 @@
     "dev:mocks": "cross-env NODE_ENV=dev-mocks vite",
     "build": "vite build",
     "type-check": "vue-tsc --build --force",
-    "lint": "eslint . --ext .vue,.js,.jsx,.cjs,.mjs,.ts,.tsx,.cts,.mts --ignore-path .gitignore",
```
Comment (johnsimons) [inline src/Frontend/package.json:12]: New version now respecs ignored files, and extension no longer needed

File: src/Frontend/package.json [hunk 5]
```diff
@@ -9,62 +9,59 @@
...
+    "vue-router": "^4.4.5",
     "vue-tippy": "^6.4.4",
     "vue-toastification": "^2.0.0-rc.5",
     "vue3-cookies": "^1.0.6",
     "vue3-simple-typeahead": "^1.0.11"
   },
   "devDependencies": {
+    "@eslint/js": "^9.10.0",
     "@pinia/testing": "^0.1.5",
-    "@rushstack/eslint-patch": "^1.10.4",
```
Comment (johnsimons) [inline src/Frontend/package.json:38]: not required

File: src/Frontend/.eslintrc.cjs (hunk 2 above)
Comment (PhilBastian) [reply src/Frontend/.eslintrc.cjs:23]: i would suggest using the deprecated eslintrc until there is a replacement for these

Comment (johnsimons) [reply src/Frontend/.eslintrc.cjs:23]: We can't, we need to move out of it.
image

And those rules are not needed, we have moved to async await now.

Comment (PhilBastian) [reply src/Frontend/.eslintrc.cjs:23]: those rules are for ensuring future work follows the async/await paradigm. The blog says that eslintrc will not be maintained, but also won't be removed. I'm not suggesting staying on eslintrc long term, just until all the rules we want to enforce are supported

Comment (johnsimons) [reply src/Frontend/.eslintrc.cjs:23]: the problem is that those are not going to be updated, so we do need to look for replacements

Comment (johnsimons) [reply src/Frontend/.eslintrc.cjs:23]: replaced with https://github.com/eslint-community/eslint-plugin-promise/blob/main/docs/rules/prefer-await-to-then.md

File: src/Frontend/.eslintrc.cjs (hunk 3 above)
Comment (johnsimons) [reply src/Frontend/.eslintrc.cjs:30]: replaced with https://eslint.org/docs/latest/rules/no-duplicate-imports

**Comment by johnsimons** [comment]
Is this ok now @PhilBastian ?

**Comment by PhilBastian** [comment]
i guess so? 🤷

**Review by PhilBastian — APPROVED** [review]

**Pull request merged**